Riot/Modules/Contacts/DataSources/ContactsDataSource.m

```diff
@@ -276,42 +284,41 @@ - (void)searchWithPattern:(NSString *)searchText forceReset:(BOOL)forceRefresh h
         }
         
         // Sort the refreshed list of the invitable contacts
-        [[MXKContactManager sharedManager] sortAlphabeticallyContacts:searchProcessingLocalContacts];
-        [[MXKContactManager sharedManager] sortContactsByLastActiveInformation:searchProcessingMatrixContacts];
+        [[MXKContactManager sharedManager] sortAlphabeticallyContacts:self->searchProcessingLocalContacts];
+        [[MXKContactManager sharedManager] sortContactsByLastActiveInformation:self->searchProcessingMatrixContacts];
+        
+        self->searchProcessingText = searchText;
         
-        searchProcessingText = searchText;
+        MXWeakify(self);
         
         dispatch_sync(dispatch_get_main_queue(), ^{
             
             // Sanity check: check whether self has been destroyed.
-            if (!searchProcessingQueue)
-            {
-                return;
-            }
+            MXStrongifyAndReturnIfNil(self);
```

Comment:
It seems to me a Weakify/Strongify would have the same effect here, but flagging in case it doesn't.